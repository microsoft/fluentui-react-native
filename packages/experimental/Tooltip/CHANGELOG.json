{
  "name": "@fluentui-react-native/tooltip",
  "entries": [
    {
      "date": "Thu, 30 Nov 2023 21:40:36 GMT",
      "version": "0.2.0",
      "tag": "@fluentui-react-native/tooltip_v0.2.0",
      "comments": {
        "minor": [
          {
            "author": "gulnazsayed@microsoft.com",
            "package": "@fluentui-react-native/tooltip",
            "commit": "b5a9fdefb58572ac6976e682563a190cb8a644b7",
            "comment": "adding new package"
          }
        ]
      }
    }
  ]
}
