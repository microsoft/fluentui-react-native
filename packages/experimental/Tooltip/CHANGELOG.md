# Change Log - @fluentui-react-native/tooltip

This log was last generated on Thu, 30 Nov 2023 21:40:36 GMT and should not be manually modified.

<!-- Start content -->

## 0.2.0

Thu, 30 Nov 2023 21:40:36 GMT

### Minor changes

- adding new package (gulnazsayed@microsoft.com)
