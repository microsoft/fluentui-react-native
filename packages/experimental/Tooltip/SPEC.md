# Tooltip

## Background

Please write here you documentation...
