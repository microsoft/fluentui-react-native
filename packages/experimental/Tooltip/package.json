{
  "name": "@fluentui-react-native/tooltip",
  "version": "0.2.0",
  "description": "A cross-platform Tooltip component for React Native.",
  "main": "src/index.ts",
  "module": "src/index.ts",
  "typings": "lib/index.d.ts",
  "onPublish": {
    "main": "lib-commonjs/index.js",
    "module": "lib/index.js"
  },
  "scripts": {
    "build": "fluentui-scripts build",
    "clean": "fluentui-scripts clean",
    "depcheck": "fluentui-scripts depcheck",
    "just": "fluentui-scripts",
    "lint": "fluentui-scripts eslint",
    "test": "fluentui-scripts jest",
    "update-snapshots": "fluentui-scripts jest -u",
    "prettier": "fluentui-scripts prettier",
    "prettier-fix": "fluentui-scripts prettier --fix true"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui-react-native.git",
    "directory": "packages/experimental/Tooltip"
  },
  "dependencies": {
    "@fluentui-react-native/callout": "^0.25.0",
    "@fluentui-react-native/component-cache": "^1.6.0",
    "@fluentui-react-native/framework": "0.12.0",
    "tslib": "^2.3.1"
  },
  "devDependencies": {
    "@fluentui-react-native/button": "^0.36.1",
    "@fluentui-react-native/eslint-config-rules": "^0.1.1",
    "@fluentui-react-native/scripts": "^0.1.1",
    "@fluentui-react-native/test-tools": ">=0.1.1 <1.0.0",
    "@office-iss/react-native-win32": "^0.72.0",
    "@react-native/metro-config": "^0.72.0",
    "react": "18.2.0",
    "react-native": "^0.72.0"
  },
  "peerDependencies": {
    "react": "18.2.0",
    "react-native": "^0.72.0"
  },
  "author": "",
  "license": "MIT",
  "rnx-kit": {
    "kitType": "library",
    "alignDeps": {
      "presets": [
        "microsoft/react-native"
      ],
      "requirements": [
        "react-native@0.72"
      ],
      "capabilities": [
        "core",
        "react"
      ]
    }
  }
}
