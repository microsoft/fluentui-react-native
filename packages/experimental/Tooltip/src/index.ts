export { Tooltip } from './Tooltip';
export { tooltipName } from './Tooltip.types';
export type { TooltipProps, TooltipSlotProps, TooltipType } from './Tooltip.types';
