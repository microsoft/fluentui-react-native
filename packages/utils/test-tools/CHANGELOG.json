{
  "name": "@fluentui-react-native/test-tools",
  "entries": [
    {
      "date": "Thu, 26 Aug 2021 04:48:40 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.3",
            "author": "67026167+chiuam@users.noreply.github.com",
            "commit": "b8c866644b2c5b64530990e59153424b0d6f5dcc",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Wed, 18 Aug 2021 23:00:57 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.2",
            "author": "ruaraki@microsoft.com",
            "commit": "345d1de930ff42359260da7f1e68a16059f4d763",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Fri, 13 Aug 2021 16:12:26 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.1",
            "author": "t-lindaweng@microsoft.com",
            "commit": "8946966df6ddc0ef1c0b43e576a4124c678d7c89",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Thu, 12 Aug 2021 17:23:44 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.0",
            "author": "ruaraki@microsoft.com",
            "commit": "e524662e3e40fe45a2dcc6e2acd3aca1197b5b5a",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Tue, 10 Aug 2021 21:12:30 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.10.0",
            "author": "ruaraki@microsoft.com",
            "commit": "ce49f33c08add7892191cf840d2fdcccfd0a16bb",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Sat, 07 Aug 2021 00:40:04 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "none": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.3",
            "author": "jasonmo@microsoft.com",
            "commit": "e506337e109ef24e675c2ac794179db86fcb7e86",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Fri, 06 Aug 2021 00:36:02 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "none": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.3",
            "author": "ruaraki@microsoft.com",
            "commit": "56ccce452d0dbbd5c24f53ebdd467f2d01dda694",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Wed, 04 Aug 2021 06:26:25 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.3",
            "author": "67026167+chiuam@users.noreply.github.com",
            "commit": "68dd61bd45ca8ffacc1580e468f3c01810e3093e",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Tue, 27 Jul 2021 22:17:20 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.2",
            "author": "jasonmo@microsoft.com",
            "commit": "1cd8611e1e88848ab66f7c0e9113c946f968260c",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Tue, 27 Jul 2021 18:55:58 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "none": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.1",
            "author": "4123478+tido64@users.noreply.github.com",
            "commit": "07d04d166b13ff79e07a8c4cbc2e06afb1e4c3e6",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Mon, 26 Jul 2021 20:41:04 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.1",
            "author": "jasonmo@microsoft.com",
            "commit": "ad02fbbe9eaa5fbe6ec6126ccb719a1280f108a6",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    }
  ]
}
