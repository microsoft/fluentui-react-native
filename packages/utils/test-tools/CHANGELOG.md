# Change Log - @fluentui-react-native/test-tools

This log was last generated on Thu, 26 Aug 2021 04:48:40 GMT and should not be manually modified.

<!-- Start content -->

## 0.1.1

Thu, 26 Aug 2021 04:48:40 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.11.3 (67026167+chiuam@users.noreply.github.com)

## 0.1.1

Wed, 18 Aug 2021 23:00:57 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.11.2 (ruaraki@microsoft.com)

## 0.1.1

Fri, 13 Aug 2021 16:12:26 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.11.1 (t-lindaweng@microsoft.com)

## 0.1.1

Thu, 12 Aug 2021 17:23:44 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.11.0 (ruaraki@microsoft.com)

## 0.1.1

Tue, 10 Aug 2021 21:12:30 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.10.0 (ruaraki@microsoft.com)

## 0.1.1

Wed, 04 Aug 2021 06:26:25 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.9.3 (67026167+chiuam@users.noreply.github.com)

## 0.1.1

Tue, 27 Jul 2021 22:17:20 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.9.2 (jasonmo@microsoft.com)

## 0.1.1

Mon, 26 Jul 2021 20:41:04 GMT

### Patches

- Bump @fluentui-react-native/theme-types to v0.9.1 (jasonmo@microsoft.com)
