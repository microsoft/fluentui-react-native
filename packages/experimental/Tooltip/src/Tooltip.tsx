/** @jsxRuntime classic */
import * as React from 'react';

import { ensureNativeComponent } from '@fluentui-react-native/component-cache';
import { mergeProps, stagedComponent } from '@fluentui-react-native/framework';

import type { TooltipProps } from './Tooltip.types';
import { tooltipName } from './Tooltip.types';

const NativeTooltipView = ensureNativeComponent('RCTTooltip');

/**
 * A function which determines if a set of styles should be applied to the component given the current state and props of the tooltip.
 *
 * @param layer The name of the state that is being checked for
 * @param userProps The props that were passed into the tooltip
 * @returns Whether the styles that are assigned to the layer should be applied to the tooltip
 */
export const tooltipLookup = (layer: string, userProps: TooltipProps): boolean => {
  return userProps[layer];
};

export const Tooltip = stagedComponent((props: TooltipProps) => {
  return (rest: TooltipProps, children: React.ReactNode) => {
    return <NativeTooltipView {...mergeProps(props, rest)}>{children}</NativeTooltipView>;
  };
});

Tooltip.displayName = tooltipName;

export default Tooltip;
