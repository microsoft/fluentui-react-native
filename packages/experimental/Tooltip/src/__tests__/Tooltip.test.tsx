import * as React from 'react';

import { ButtonV1 } from '@fluentui-react-native/button';
import { checkRenderConsistency, checkReRender } from '@fluentui-react-native/test-tools';
import * as renderer from 'react-test-renderer';

import { Tooltip } from '../Tooltip';

describe('Tooltip component tests', () => {
  it('Tooltip default', () => {
    const tree = renderer
      .create(
        <Tooltip content="Your component">
          <ButtonV1>Your component</ButtonV1>
        </Tooltip>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tooltip simple rendering does not invalidate styling', () => {
    checkRenderConsistency(
      () => (
        <Tooltip content="Default Tooltip">
          <ButtonV1>Default Tooltip</ButtonV1>
        </Tooltip>
      ),
      2,
    );
  });

  it('Tooltip re-renders correctly', () => {
    checkReRender(
      () => (
        <Tooltip content="Render twice">
          <ButtonV1>Render twice</ButtonV1>
        </Tooltip>
      ),
      2,
    );
  });

  // Feel free to add more tests here
});
