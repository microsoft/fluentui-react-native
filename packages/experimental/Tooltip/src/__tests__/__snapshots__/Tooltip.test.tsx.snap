// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component tests Tooltip default 1`] = `
<RCTTooltip
  content="Your component"
>
  <View
    accessibilityLabel="Your component"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "multiselectable": undefined,
        "required": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    enableFocusRing={true}
    focusable={true}
    keyUpEvents={
      [
        {
          "key": " ",
        },
        {
          "key": "Enter",
        },
      ]
    }
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onLayout={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#ffffff",
        "borderColor": "#d1d1d1",
        "borderRadius": 4,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "minHeight": 24,
        "minWidth": 64,
        "padding": 3,
        "paddingHorizontal": 7,
        "width": undefined,
      }
    }
  >
    <Text
      accessible={false}
      ellipsizeMode="tail"
      numberOfLines={0}
      onAccessibilityTap={[Function]}
      style={
        {
          "color": "#242424",
          "fontFamily": "Segoe UI",
          "fontSize": 12,
          "fontWeight": "400",
          "margin": 0,
          "marginBottom": 1,
          "marginEnd": -2,
          "marginStart": 0,
          "marginTop": -1,
        }
      }
    >
      Your component
    </Text>
  </View>
</RCTTooltip>
`;
